engine/menu/field_debug_menu.asm

```diff
@@ -0,0 +1,518 @@
+	call PlaySFX
+.loop
+	call GetJoypad
+	ldh a, [hJoyDown]
+	bit 0, a
+	jr nz, .loop
+	ld a, $00
+	ret
+
+CloseFieldDebugMenuTextBoxLoop: ; 3F:41A1
+	push bc
+	ld b, a
+.loop
+	call GetJoypad
+	ld a, [hJoyDown]
+	and b
+	jr z, .loop
+	pop bc
+	ret
+
+ShowTextBoxAndExitFieldDebugMenu: ; 3F:41AD
+	call MenuTextBox
+	ld a, $03
+	call CloseFieldDebugMenuTextBox
+	call CloseWindow
+	ret
+
+FieldDebug_ChangeWindowBorder: ; 3F:41B9
+	call ChangeWindowBorderMenu
+	ld a, $00
+	ret
+
+FieldDebug_ResetPrompt: ; 3F:41BF
+	call ResetPromptMenu
+	ld a, $00
+	ret
+
+Unreferenced_ShowTrainerCard ; 3F:41C5
+	ld hl, $5F25 ; 04:5F25 = Trainer Card display subroutine
+	ld a, $04
+	call FarCall_hl
+	ld a, $00
+	ret
+
+FieldDebug_ChangeTransportation: ; 3F:41D0
+	ld hl, ChangeTransportationMenuHeader
+	call LoadMenuHeader
+	ld a, [wPlayerState]
+	call GetCurrentTransportation
+	ld [wMenuCursorBuffer], a
+	dec a
+	call CopyNameFromMenu
+	call VerticalMenu
+	jp c, .exit
+	ld a, [w2DMenuDataEnd]
+	call SetCurrentTransportation
+	ld hl, wPlayerState
+	cp [hl]
+	jr z, .exit
+	cp $04
+	jr z, .check_surf
+	ld [wPlayerState], a
+	and a
+	jr z, .walking
+.biking
+.skateboard
+	ld a, [w2DMenuDataEnd]
+	dec a
+	call CopyNameFromMenu
+	call CloseWindow
+	call CloseWindow
+	ld hl, TransportationPlayerString2
+	call MenuTextBox
+	jr .update_sprite
+.walking
+	ld a, $ff
+	ld [wcc9a], a
+	call CloseWindow
+	call CloseWindow
+	call PlayMapMusic
+	ld hl, TransportationPlayerString1
+	call MenuTextBox
+.update_sprite
+	ld hl, GetPlayerSprite
+	ld a, $05
+	call FarCall_hl
+	ld a, $0f
+	call CloseFieldDebugMenuTextBoxLoop
+	call CloseWindow
+	ld a, $03
+	ret
+.check_surf
+	call CheckIfFacingSurfableTile
+	jr c, .cannot_surf
+	ld [wPlayerState], a
+	call Function_fc2ee
+	ld a, [w2DMenuDataEnd]
+	dec a
+	call CopyNameFromMenu
+	call CloseWindow
+	call CloseWindow
+	ld hl, TransportationPlayerString2
+	call MenuTextBox
+	jr .update_sprite
+.cannot_surf
+	ld hl, TransportationCannotSurfString
+	call MenuTextBox
+	ld a, $0f
+	call CloseFieldDebugMenuTextBoxLoop
+	call CloseWindow
+.exit
+	call CloseWindow
+	ld a, $00
+	ret
+
+ChangeTransportationMenuHeader:: ; 3F:426F
+	db MENU_BACKUP_TILES ; flags
+	menu_coords 03, 03, SCREEN_WIDTH - 8, SCREEN_HEIGHT - 5
+	dw .TransportationMenuData
+	db 01 ; default option
+
+.TransportationMenuData: ; 3F:4277
+	db STATICMENU_CURSOR | STATICMENU_WRAP ; flags
+	db 04 ; items
+	db "あるき@"
+	db "じてんしゃ@"
+	db "スケボー@"
+	db "ラプラス@"
+
+TransportationPlayerString1: ; 3F:428D
+	text "<PLAYER>は@"
+
+TransportationPlayerString2: ; 3F:429D
+	text "<PLAYER>は@"
+
+TransportationCannotSurfString: ; 3F:42AC
+	text "ここでは　のることが"
+	next "できません"
+	prompt
+
+SetCurrentTransportation: ; 3F:42BE
+	ld hl, TransportationOptions
+	dec a
+	ld c, a
+	ld b, $00
+	add hl, bc
+	ld a, [hl]
+	ret
+
+GetCurrentTransportation: ; 3F:42C8
+	ld hl, TransportationOptions
+	ld b, $01
+.loop
+	cp [hl]
+	jr z, .return
+	inc hl
+	inc b
+	jr .loop
+.return
+	ld a, b
+	ret
+
+WALKING    EQU 0
```

Comment:
If you have file-specific constants, define them at the top of the file, before any `SECTION`. That way they're easier to find if someone ever needs to move them to the global constants files instead.